src/com/ubcsolar/sim/SimController.java

```diff
@@ -115,6 +115,110 @@ public void runSimulation(int laps, long startTime)
+					//for each runSimV2 call. We assume that if the cannot make it through without conserving charge for the next runSimV2 call (giving a non-zero
+					//value for min charge), it will not make it through if charge is conserved (may need to change for single lap routes, definitely needs tweaking for
+					//for multiple laps)
+					result = new SimEngine().runSimV2(routeToTraverse, start, end,
+							lastCarReported, manualSpeedProfile.get(start).get(i), manualSpeedProfile.get(end).get(i),
+							nextStartTime, i, 0, current_forecast);
+					//if sim is not successful, remove all the points before the end of the sim (only want to graph to the point the car fails, so speeds after 
+					//that needs to be removed in speed profile)
+					if(!result.wasRunSuccessful()) {
+						System.err.println("Car cannot make it through the route with selected speeds");
+						for (int removed_point = routeToTraverse.getTrailMarkers().size() - 1; (!end.equals(removed_point) && removed_point >= 0);removed_point--) {
+							manualSpeedProfile.remove(routeToTraverse.getTrailMarkers().get(removed_point));
+						}
+						break;
+					}
+					//if sim is successful, update parameters
+					if (point_index < routeToTraverse.getTrailMarkers().size() - 1) {
+						start = routeToTraverse.getTrailMarkers().get(++point_index);
+					}
+					lastCarReported = result.getFinalTelemData();
+					resultFrames.addAll(result.getListOfFrames()); // add sim frames to list
+	                nextStartTime += result.getTravelTime();
+	                totalTime += result.getTravelTime();
+				}
+			}
+			if (!result.wasRunSuccessful()) {
+				break;
+			}
+		}
+			/*
+			//after simulating almost the whole route, check to see if simulations were successful
+			//if not, print error message 
+			if (!result.wasRunSuccessful()) {
+				System.err.println("Car cannot make it through the route with selected speeds");
+				break;
+			}
+			//if everything has been successful so far, run one last simulation to the end of the route
+			//this cannot be done in the previous for loop since the loop only runs simulations if there is a speed change
+			//most, if not all of the time, the speed would be constant from one point of the route to the end, and this 
+			//will not trigger a runSimV2 call in the for loop
+			else {
+				result = new SimEngine().runSimV2(routeToTraverse, start, routeToTraverse.getTrailMarkers().get(routeToTraverse.getTrailMarkers().size() - 1),
+						lastCarReported, manualSpeedProfile.get(start).get(i), 
+						manualSpeedProfile.get(routeToTraverse.getTrailMarkers().get(routeToTraverse.getTrailMarkers().size()-1)).get(i),
+						nextStartTime, i, 0, current_forecast);
+				
+				lastCarReported = result.getFinalTelemData();
+				resultFrames.addAll(result.getListOfFrames()); // add sim frames to list
+	            totalTime += result.getTravelTime();
+			}*/
```

Comment:
if you don't think we need this commented out code anymore, please remove it. If we need it back we can go through our git history